server/src/main/java/org/opensearch/index/translog/BufferedChecksumStreamInput.java

```diff
@@ -120,7 +120,7 @@ public void reset() throws IOException {
     public int read() throws IOException {
         try {
             return readByte() & 0xFF;
-        } catch (EOFException e) {
+        } catch (EOFException | ArrayIndexOutOfBoundsException e) {
```

Comment:
Why would ArrayIndexOutOfBoundsException occur?

Reply:
When the read of the byte array is completed, AIOB is thrown and not EOF .

Reply:
Got it . I have fixed `BytesStreamInput` now to return `EOF` instead of `AIOOBE` just like other implementations of `StreamInput` .